code/modules/assembly/flash.dm

```diff
@@ -81,6 +81,8 @@
 /obj/item/device/assembly/flash/proc/get_flash_targets(atom/target_loc, range = 3, override_vision_checks = FALSE)
 	if(!target_loc)
 		target_loc = loc
+		if(!isturf(target_loc))
+			target_loc = loc.loc

```

Comment:
so the flash is actually contained in an assembly holder object meaning its loc is not a turf or something and as far as i can tell it breaks it. the AOE flash happens but if youre standing next to it you're not affected by it. I THINK this is the part of it that is broken. and what i've added here didn't seem to fix it, either.

Reply:
use get_turf